view/src/components/farms.js

```diff
@@ -0,0 +1,819 @@
+                      defaultValue={this.state.farmTags}
+                      freeSolo
+                      renderTags={(value, getTagProps) =>
+                        value.map((option, index) => (
+                          <Chip label={option} {...getTagProps({ index })} />
+                        ))
+                      }
+                      renderInput={(params) => (
+                        <TextField
+                          {...params}
+                          variant="outlined"
+                          label="Farm Tags"
+                          placeholder="tags..."
+                        />
+                      )}
+                    />
+                  </Grid>
+                </Grid>
+              </form>
+            </Container>
+          </Dialog>
+          <Container maxWidth="lg">
+            <Grid container spacing={2} alignItem="center">
+              <Grid item xs={12}>
+                <div className={classes.search}>
+                  <div className={classes.searchIcon}>
+                    <SearchIcon />
+                  </div>
+                  <InputBase
+                    fullWidth={true}
+                    placeholder="Search…"
+                    classes={{
+                      root: classes.inputRoot,
+                      input: classes.inputInput,
+                    }}
+                    inputProps={{ "aria-label": "search" }}
+                  />
+                </div>
+              </Grid>
+              {this.state.farms.map((farm) => (
+                <Grid item xs={12}>
+                  <Card className={classes.root} variant="outlined">
+                    <CardContent>
+                      <Typography variant="h5" component="h2">
+                        {farm.farmName}
+                      </Typography>
+                      {farm.farmTags.map((tag) => (
+                        <Chip
+                          className={classes.chip}
+                          label={tag}
+                          size="small"
+                        />
+                      ))}
+                      <Box
+                        display="flex"
+                        flexDirection="row"
+                        flexWrap="wrap"
+                        p={0}
+                        m={0}
+                      >
+                        <Box p={3}>
+                          <Typography
+                            className={classes.pos}
+                            color="textSecondary"
+                          >
+                            Details:
+                          </Typography>
+                          <Typography variant="body2" component="p">
+                            Location of Farm:
+                            {`${farm.location.substring(0, 30)}`}
+                            <br />
+                            {`${farm.location.substring(30, 78)}`}
```

Comment:
What is this about?  Seems very odd to assume that the location of every farm is exactly 78 characters long and can be broken at column 30.  Are you trying to split the string on a particular part of the address?  Or just make it wrap?

Reply:
Since the grid expands to any size depending on the content, the wrapping CSS was't working. So I was intending just to show a snipped of the location and truncate it if its too long. Thoughts?

Reply:
Hmm, are you saying that the grid cell becomes too wide then, because there aren't any line breaks in the address?  There are probably ways to fix that, but I'd need to see the CSS of the grid to know for sure.  You might be able to wrap it in an element with a `max-width` to prevent the grid cell from growing without bound.

Reply:
Awesome! I'll check it out right now